scripts/name_match.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python
+
+import os
+import glob
+import click
+import csv
+from collections import defaultdict
+from utils import get_filename, get_data_dir, load_yaml, dump_obj
+
+unmatched = []
+
+
+def add_to_file(name_to_add, person_file):
+    other_names_found = False
+    same_name_found = False
+    try:
+        with open(person_file, "r") as file:
+            file_text = file.read()
+            if "other_names:" in file_text:
+                other_names_found = True
+                if name_to_add in file_text:
+                    same_name_found = True
+
+        with open(person_file, "a") as file:
+            if same_name_found == False:
+                if other_names_found:
+                    file.write(f"\n - name: {name_to_add}")
+                else:
+                    file.write("other_names:\n")
+                    file.write(f" - name: {name_to_add}")
+    except OSError as err:
+        pass
+
+
+def interactive_check(csv_name, yml_name, possible_name_match):
+
+    choices = ""
+    if possible_name_match:
+        click.secho(f"Last name matched between '{csv_name}' and '{yml_name}'", fg="yellow")
+        text = "(a)dd name?"
+
+    ch = "~"
+    if possible_name_match:
+        choices = "a"
+
+    while ch not in (choices + "sa"):
+        click.secho(text + " (s)kip? (e)xit?", bold=True)
+        ch = click.getchar()
+
+    if ch == "e":
+        raise SystemExit(-1)
+    elif ch == "a":
+        click.secho(" adding.", fg="green")
+    elif ch == "s":
+        click.secho(" skipping.", fg="green")
+        return False
+
+    return True
+
+
+def find_match(name, person):
+
+    possible_name_match = False
+    matched = False
+
+    cleaned_name = str.lower(name)
+    cleaned_name = cleaned_name.replace(".", " ")
+    cleaned_name = cleaned_name.replace("*", " ")
+    cleaned_name = cleaned_name.replace(",", " ")
+
+    cleaned_person_name = str.lower(person["name"])
+    cleaned_person_name = cleaned_person_name.replace(".", " ")
+    cleaned_person_name = cleaned_person_name.replace("*", " ")
+    cleaned_person_name = cleaned_person_name.replace(",", " ")
+
+    if person.get("family_name") != None:
+
+        cleaned_person_family_name = str.lower(person["family_name"])
+
+        if cleaned_name == cleaned_person_family_name:
+            matched = True
+        elif cleaned_name in cleaned_person_family_name:
```

Comment:
this is probably too aggressive, I'd at least ensure that the cleaned_name doesn't just appear in the middle of another word